### ISSUE TITLE
Merge User Guide into Cookbook

### ISSUE THREAD

**Issue by vkostyukov** [finagle/finch#665]
As far as OSS docs, /I think/ I much prefer a cookbook (covers a little of each concept in a single standalone example/recipe) over the user-guide (covers each concept separately, step-by-step) styles.

I sort of find the current version of the user guide too verbose, carrying lots irrelevant details. I tend to believe that the combination of a rich cookbook (solves particular problems; gives initial impression) and readable source code (gives in-depth/implementation details) would give the community much more than the current user guide.

What do people think about this?

**Comment by spockz** [comment]
We have debated this same issue within our team as well because we noticed people coming to the docs with concrete issues to solve. Sometimes experienced developers and sometimes beginners. Both benefit from the recipe style but it also stimulates copy and paste programming which leads to less understanding and more questions about simple situations that are just a bit different from the recipe.

On the other hand the user guide is quite detailed and sometimes rather dry. This makes it possible to quickly get to thoroughly learn the framework from the bottom up but doesn't help when you just want to build a simple app.

This is why we decided on a best of both worlds: detailed user guides with details on how and why things work and recipes for things we see often.

Maintaining two distinct sets of documentation is quite a bit of work but I think it pays of in the end.

Personally, I really like the user guides with lots of details and explanations on why things are as they are on a conceptual level because it helps me think of how the library work.
